### PR TITLE
[Major] ESIL to REIL Converter v0

### PR DESCRIPTION
**Pull request by sushant94**
ESIL to REIL Translator.
Pull Request for issue #2190

### CODE REVIEW

**Comment by jvoisin** [comment]
Awesome!

File: libr/anal/esil2reil.c [hunk 1]
```diff
@@ -0,0 +1,1124 @@
+/*
+ * Convert from ESIL to REIL (Reverse Engineering Intermediate Language)
+ * Contributor: sushant94
+ */
+
+#include <r_anal.h>
+
+#define REIL_TEMP_PREFIX "V"
+void reil_flag_spew_inst(RAnalEsil *esil, const char *flag);
+const char *ops[] = { FOREACHOP(REIL_OP_STRING) };
```
Comment (jvoisin) [inline libr/anal/esil2reil.c]: Make it static.

File: libr/anal/esil2reil.c [hunk 2]
```diff
@@ -0,0 +1,1124 @@
...
+	if (!i)
+		return R_FALSE;
+	return (ut8)i->size;
+}
+
+RAnalReilArgType reil_get_arg_type(RAnalEsil *esil, char *s) {
+	if (!strncmp(s, REIL_TEMP_PREFIX, strlen(REIL_TEMP_PREFIX)))
+		return ARG_TEMP;
+
+	int type = r_anal_esil_get_parm_type(esil, s);
```
Comment (jvoisin) [inline libr/anal/esil2reil.c]: What about making it `const` ?

File: libr/anal/esil2reil.c [hunk 3]
```diff
@@ -0,0 +1,1124 @@
...
+		strncpy(op->name, tmp_buf, sizeof(op->name));
+		op->type = reil_get_arg_type(esil, op->name);
+		if (op->type == ARG_REG) 
+			op->size = esil_internal_sizeof_reg(esil, op->name);
+		else if (op->type == ARG_CONST)
+			op->size = esil->anal->bits;
+		return op;
+	}
+
+	op->size = atoi(tmp_buf);
```
Comment (jvoisin) [inline libr/anal/esil2reil.c]: Shall we use `op->size =  strtol(tmp_buf, NULL, 10);` instead?
See `man 3 atoi` for details.

**Comment by jvoisin** [comment]
What about a `reil_free_inst`, since there is `reil_new_inst`, and there are multiple fields to free.

File: libr/anal/esil2reil.c (hunk 3 above)
Comment (condret) [inline libr/anal/esil2reil.c]: strtol (tmp_buf, NULL, 16); pls if possible

File: libr/anal/esil2reil.c [hunk 4]
```diff
@@ -0,0 +1,1124 @@
...
+// Get the next available temp register.
+void get_next_temp_reg(RAnalEsil *esil, char *buf) {
+	int n;
+	n = esil->Reil->reilNextTemp;
+	snprintf(buf, sizeof(buf), "%s_%02d", REIL_TEMP_PREFIX, n);
+	esil->Reil->reilNextTemp++;
+}
+
+void reil_make_arg(RAnalEsil *esil, RAnalReilArg *arg, char *name) {
+	if (!arg)
```
Comment (crowell) [inline libr/anal/esil2reil.c]: return can go on the same

File: libr/anal/esil2reil.c [hunk 5]
```diff
@@ -0,0 +1,1017 @@
...
+RAnalReilArg *reil_pop_arg(RAnalEsil *esil) {
+	RAnalReilArg *op;
+	int i, j = 0, flag = 0, len;
+	char tmp_buf[32];
+	char *buf = r_anal_esil_pop(esil);
+	if (!buf) return NULL;
+
+	len = strlen(buf);
+	op = R_NEW0(RAnalReilArg);
+	for (i = 0; i < len; i++) {
```
Comment (crowell) [inline libr/anal/esil2reil.c]: this can be done easier with `strchr` probably.

File: libr/anal/esil2reil.c [hunk 6]
```diff
@@ -0,0 +1,1017 @@
...
+	if (!ins) return;
+	if (ins->arg[0]) R_FREE(ins->arg[0]);
+	if (ins->arg[1]) R_FREE(ins->arg[1]);
+	if (ins->arg[2]) R_FREE(ins->arg[2]);
+	R_FREE(ins);
+}
+
+// Automatically increments the seq_num of the instruction.
+void reil_print_inst(RAnalEsil *esil, RAnalReilInst *ins) {
+	if (!ins) return;
```
Comment (crowell) [inline libr/anal/esil2reil.c]: also check null of `esil`?

File: libr/anal/rpnesil.c [hunk 7]
```diff
@@ -2665,6 +2688,9 @@ static int runword (RAnalEsil *esil, const char *word) {
 		// skip empty words
 		return 1;
 	}
+
+
```
Comment (crowell) [inline libr/anal/rpnesil.c]: rm 2 of these empty lines

File: libr/anal/rpnesil.c [hunk 8]
```diff
@@ -2651,6 +2672,8 @@ static int runword (RAnalEsil *esil, const char *word) {
 		return 1;
 	}
 
+
+
```
Comment (crowell) [inline libr/anal/rpnesil.c]: rm 1 of these empty lines

**Comment by crowell** [comment]
LGTM :+1:

**Comment by XVilka** [comment]
Thanks! Merged in 1d89f7d7996e67ea9275fc1318aeb9390def27cc

**Pull request closed without merging**